hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/extension/MonitorServiceRestartParentAttributeHandler.java

```diff
@@ -116,8 +122,29 @@ protected boolean applyUpdateToRuntime(OperationContext context, ModelNode opera
             }
         }
 
-        return super.applyUpdateToRuntime(context, operation, attributeName, resolvedValue, currentValue,
-                handbackHolder);
+        // When updating agent config we by default allow immediate restarts, the caller does not have to be aware
+        // of which attributes may or may not force the restart.  In some situations, namely updates made by
+        // an agent command like UpdateCollectionIntervalsCommand, we need to defer the restart because otherwise
+        // the command response will never be sent back.  To defer the restart the operation must explicitly
+        // have the header ALLOW_RESOURCE_SERVICE_RESTART=false (note, each operation in a composite must assign
+        // it individually). In this situation not only do we want to defer the restart, we want to avoid setting
+        // the server to 'reload-required'.  To do this we temporarily mark the parent as restarted, and we must
+        // also return false from this method.
+        ModelNode headers = operation.has(OPERATION_HEADERS) ? operation.get(OPERATION_HEADERS) : null;
+        boolean restartAllowedOptionSet = (headers != null) && headers.hasDefined(ALLOW_RESOURCE_SERVICE_RESTART);
+        boolean restartAllowed = !restartAllowedOptionSet || headers.get(ALLOW_RESOURCE_SERVICE_RESTART).asBoolean();
+        Object tempOwner = null;
+        PathAddress address = null;
+        if (!restartAllowed) {
+            address = getParentAddress(context.getCurrentAddress());
```

Comment:
Sorry missed this one before, we don't have this method on EAP6 `OperationContext.getCurrentAddress`
Please use [this](https://github.com/hawkular/hawkular-agent/blob/master/hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/util/WildflyCompatibilityUtils.java#L195) one instead.
`WildflyCompatibilityUtils.getCurrentAddress(context, operation)`
Note that on EAP6 the address is taken from the operation.